internal/watcher/health/health_watcher_service.go

```diff
@@ -131,7 +151,25 @@ func (hw *HealthWatcherService) updateCache(currentHealth map[string]*v1.Instanc
 	slog.Debug("Updating health watcher cache", "cache", hw.cache)
 }
 
-func (hw *HealthWatcherService) compareHealth(currentHealth map[string]*v1.InstanceHealth) bool {
+func (hw *HealthWatcherService) compareCache(healthStatuses []*mpi.InstanceHealth) []*mpi.InstanceHealth {
+	if len(hw.cache) != len(hw.instances) {
+		for instanceID := range hw.cache {
+			if _, ok := hw.instances[instanceID]; !ok {
+				health := &mpi.InstanceHealth{
+					InstanceId:           instanceID,
+					InstanceHealthStatus: mpi.InstanceHealth_INSTANCE_HEALTH_STATUS_UNHEALTHY,
+					Description:          fmt.Sprintf("instance has been deleted %s", instanceID),
```

Comment:
```suggestion
					Description:          fmt.Sprintf("instance %s not found", instanceID),
```

I think not found might be a better way to word it than saying it was deleted